### PR TITLE
[NFC][Build System: build-script] Move the static methods from the BuildScriptInvocation class to the top level of build-script.

### PR DESCRIPTION
**Pull request by Rostepher**

### CODE REVIEW

**Comment by Rostepher** [comment]
@swift-ci please test

File: utils/build-script [hunk 1]
```diff
@@ -131,192 +152,192 @@ def tar(source, destination):
...
+    if host_target == StdlibDeploymentTarget.OSX.x86_64:
+        targets = [host_target]
+        if args.build_ios and args.build_ios_simulator:
+            targets.extend(StdlibDeploymentTarget.iOSSimulator.targets)
+        if args.build_ios and args.build_ios_device:
+            targets.extend(StdlibDeploymentTarget.iOS.targets)
+        if args.build_tvos and args.build_tvos_simulator:
+            targets.extend(StdlibDeploymentTarget.AppleTVSimulator.targets)
+        if args.build_tvos and args.build_tvos_device:
+            targets.extend(StdlibDeploymentTarget.AppleTV.targets)
```
Comment (compnerd) [inline utils/build-script]: Similar for tvOS

**Comment by Rostepher** [comment]
@compnerd This change is intended as a completely NFC in which I am merely moving around code I want to start touching. While I appreciate the comments on functionality and best practices I would like to merge this PR as-is to keep the content of future PRs clear.

**Comment by compnerd** [comment]
Okay, I think that we should still do those cleanups in a follow up change then.

**Review by compnerd — APPROVED** [review]

**Comment by Rostepher** [comment]
I'm definitely happy to do all of the clean-ups you've suggested, just in future PRs.

**Pull request merged**